File: output/xCodeEval_remove_8000/C++/Python/40ca10a54463a16f6c0559d2d974fdc5.py
import math
import array

N = 18000
M = 18000
is_prime = array.array('b', [True] * N)
prm = array.array('i')
cntp = 0

def id0(n):
    global cntp
    prm.append(2)
    is_prime[0] = is_prime[1] = False
    for i in range(4, n, 2):
        is_prime[i] = False
    for i in range(3, int(math.sqrt(n)) + 1, 2):
        if is_prime[i]:
            prm.append(i)
            for j in range(i * 2, n, i):
                is_prime[j] = False
    for i in range(i, n, 2):
        if is_prime[i]:
            prm.append(i)
    cntp = len(prm)
    return cntp

File: output/GPT4o-Mini/xCodeEval_4000/C++/Java/40ca10a54463a16f6c0559d2d974fdc5.java
import java.util.Arrays;
import java.util.Scanner;

public class 40ca10a54463a16f6c0559d2d974fdc5_nc {
    static final int N = 18000;
    static final int M = 18000;
    static boolean[] is = new boolean[N];
    static int[] prm = new int[M];
    static int cntp;

    static int getprm(int n) {
        int k = 0;
        int e = (int) (Math.sqrt(n) + 1);
        Arrays.fill(is, true);
        prm[k++] = 2;
        is[0] = is[1] = false;
        for (int i = 4; i < n; i += 2) is[i] = false;
        for (int i = 3; i < e; i += 2) {
            if (is[i]) {
                prm[k++] = i;
                for (int j = i * i; j < n; j += i * 2) {
                    is[j] = false;
                }
            }
        }
        for (int i = e; i < n; i += 2) {
            if (is[i]) prm[k++] = i;
        }
        return k;
    }
